code/game/objects/items/weapons/extinguisher.dm

```diff
@@ -158,4 +158,13 @@
 					sleep(2)
 
 	else
-		return ..()
\ No newline at end of file
+		return ..()
+
+/obj/item/weapon/extinguisher/AltClick(mob/user)
+	EmptyExtinguisher(user)
+
+/obj/item/weapon/extinguisher/proc/EmptyExtinguisher(var/mob/user)
+	if(loc == user)
+		reagents.clear_reagents()
+		user << "<span class='info'>You quietly empty out the [src].</span>"

```

Comment:
I think this should be a visible message. It's not exactly easy to dump that much water out of a highly-pressurised tank with no release valve.